src/Service/AvailableCarService.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Service;
+
+use App\Entity\Booking;
+use App\Entity\Car;
+use App\Enum\EnumBookingStatus;
+use App\Service\PricingApi\PricingClientInterface;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Contracts\Cache\CacheInterface;
+use Symfony\Contracts\Cache\ItemInterface;
+
+class AvailableCarService
+{
+    public function __construct(
+        private readonly EntityManagerInterface $entityManager,
+        private readonly PricingClientInterface $pricingClient,
+        private readonly CacheInterface $cache,
+    ) {
+    }
+
+    public function getAvailableCars(
+        int $stationId,
+        \DateTimeInterface $startDate,
+        \DateTimeInterface $endDate,
+    ): array {
+        return $this->cache->get(
+            $this->getCacheKey($stationId, $startDate, $endDate),
+            function (ItemInterface $item) use ($stationId, $endDate, $startDate): array {
+                $item->expiresAfter(300); // Cache for 5 minutes
+
+                return $this->getAvailableCarsInternal($stationId, $startDate, $endDate);
+            }
+        );
+    }
+
+    private function getCacheKey(int $stationId, \DateTimeInterface $startDate, \DateTimeInterface $endDate): string
+    {
+        return md5(
+            \sprintf(
+                '%s_%s_%s_%s',
+                __CLASS__,
+                $stationId,
+                $startDate->format(\DateTimeInterface::ISO8601_EXPANDED),
+                $endDate->format(\DateTimeInterface::ISO8601_EXPANDED)
+            )
+        );
+    }
+
+    private function getAvailableCarsInternal(int $stationId, \DateTimeInterface $startDate, \DateTimeInterface $endDate)
+    {
+        $cars = $this->entityManager->getRepository(Car::class)->findBy(
+            [
+                'active' => true,
+            ]
+        );
+
+        $availableCars = [];
+
+        foreach ($cars as $car) {
+            if ($car->getStation()?->getId() !== $stationId) {
```

Comment:
```suggestion
            if ($car->getStation()?->getId() != $stationId) {
```
Too strict  :nerd_face: